Corona-Warn-App/src/main/java/de/rki/coronawarnapp/appconfig/mapping/SurveyConfigMapper.kt

```diff
@@ -5,17 +5,46 @@ import de.rki.coronawarnapp.appconfig.SafetyNetRequirements
 import de.rki.coronawarnapp.appconfig.SafetyNetRequirementsContainer
 import de.rki.coronawarnapp.appconfig.SurveyConfig
 import de.rki.coronawarnapp.server.protocols.internal.v2.AppConfigAndroid
+import timber.log.Timber
 import javax.inject.Inject
 
 @Reusable
 class SurveyConfigMapper @Inject constructor() : SurveyConfig.Mapper {
     override fun map(rawConfig: AppConfigAndroid.ApplicationConfigurationAndroid): SurveyConfig {
-        // TODO
-        return SurveyConfigContainer()
+        var surveyConfigContainer = SurveyConfigContainer()
+
+        if (rawConfig.hasEventDrivenUserSurveyParameters()) {
+            if (rawConfig.eventDrivenUserSurveyParameters.hasCommon()) {
+                rawConfig.eventDrivenUserSurveyParameters.common.also {
+                    surveyConfigContainer = SurveyConfigContainer(
+                        otpQueryParameterName = it.otpQueryParameterName,
+                        surveyOnHighRiskEnabled = it.surveyOnHighRiskEnabled,
+                        surveyOnHighRiskUrl = it.surveyOnHighRiskUrl
+                    )
+                }
+            }
+
+            if (rawConfig.eventDrivenUserSurveyParameters.hasPpac()) {
+                rawConfig.eventDrivenUserSurveyParameters.ppac.also {
+                    surveyConfigContainer = surveyConfigContainer.copy(
+                        safetyNetRequirements = SafetyNetRequirementsContainer(
+                            requireBasicIntegrity = it.requireBasicIntegrity,
+                            requireCTSProfileMatch = it.requireCTSProfileMatch,
+                            requireEvaluationTypeBasic = it.requireEvaluationTypeBasic,
+                            requireEvaluationTypeHardwareBacked = it.requireEvaluationTypeHardwareBacked
+                        )
+                    )
+                }
+            }
```

Comment:
Is it possible that we have only one of the parameters ( `common` / `ppac`)  and not the other?
Should we throw an error or use a fallback if both are not available?

Reply:
If e.g. `ppac` is optional what should the default values be for survey related ppac?

Reply:
would it make sense to have an own mapper for PPAC too? because it is also needed at analytics